### DISCUSSION TITLE
How to use Blameable Behaviour if activeRecord is created by Queue Job?

### DISCUSSION THREAD

**Discussion by gb5256** [yiisoft/yii2#19845] [Q&A] [answered]
I am using the Blameable Behaviour to log who creates/updated records.
But I am now implementing a function to update records via a Job coming from the Queue.
The Queue is running from the Console (I am using the AdvancedTemplate).
But by doing this, the user_id is empty on console, so the blameable bahaviour is using NULL for the user_id for the created_at and updated_at attributes of that model.

What I try to archvie is: I edit the job so that it also holds the user_id of the user creating the job. But how can I use this info in the console?
Is there a way to tell the console to run with a certain user_id? I guess that I have to do that inside the Job (as it runs from the console), but I am lost on how to do that.

Thanks for any ideas!
gb5256

**Comment by tomaszkane** [comment]
When app has `user`
https://github.com/yiisoft/yii2/blob/master/framework/behaviors/BlameableBehavior.php#LL104C63-L104C63
then `defaultValue` is used
https://github.com/yiisoft/yii2/blob/master/framework/behaviors/BlameableBehavior.php#LL107C55-L107C55

Maybe useful?

Other way is to mock `userId` by our queue job.

**Comment by gb5256** [comment]
Hi,
yes I saw the defaultValue, but can I access (and change) that attribute of the Behaviour in runtime? 
Like based on the user_id in the job, set the defaultValue of the bahaviour?

**Comment by gb5256** [comment, marked as answer]
After more digging I am using this approach now:

In console/config/main adding this to the components:

    'user' => [
            'class' => 'yii\web\User',
            'identityClass' => 'common\models\User',
        ],

then inside the job we can do this:

    $user = User::findIdentity($this->user_id);
    Yii::$app->user->setIdentity($user);

'setIdentity' does not create session and also does not use Authentification etc. So for this user case perfect.

Thanks @tomaszkane for thinking with me on solutions.
